source/Plugins/ExpressionParser/Swift/SwiftExpressionParser.cpp

```diff
@@ -2020,7 +2039,37 @@ unsigned SwiftExpressionParser::Parse(DiagnosticManager &diagnostic_manager,
   // list of loaded modules, and copy the Decls that were globalized
   // as part of the parse from the staging area in the external
   // lookup object into the SwiftPersistentExpressionState.
-  swift::ModuleDecl *module = &parsed_expr->module;
+  // SWIFT_ENABLE_TENSORFLOW: 
+  swift::ModuleDecl *module = nullptr;
+  if (!swift_ast_ctx->GetReplExprModulesDir()) {
+    // Just reuse the module if no serialization is requested.
+    module = &parsed_expr->module;
+  } else {
+    // Reload the serialized module so that we can look
+    // into SIL functions in subsequent cells if needed (e.g., differentiation).
+    if (log) {
+      log->Printf("Reloading the serialized module.\n");
+    }
+
+    lldb::StackFrameSP this_frame_sp(m_stack_frame_wp.lock());
+
+    if (this_frame_sp) {
+      lldb::ProcessSP process_sp(this_frame_sp->CalculateProcess());
+      Status error;
+      std::string module_name = parsed_expr->module.getName().str();
+      if (process_sp && !module_name.empty()) {
+        ConstString module_const_str(module_name);
+        module = swift_ast_ctx->FindAndLoadModule(module_const_str,
```

Comment:
Is there a fundamental reason why it has to be stored into a file, or is it that all the functions we need like `FindAndLoadModule` only operate on files?

Reply:
Yes. [SerializedSILLoader::lookupSILFunction](https://github.com/apple/swift/blob/9fa91e97aa98990c78cf2595960bb40cf0772abc/lib/Serialization/SerializedSILLoader.cpp#L48) only searches for a function in serialized SIL modules in a given AST context. The serialized SIL modules are stored in [LoadedSILSections](https://github.com/apple/swift/blob/9fa91e97aa98990c78cf2595960bb40cf0772abc/lib/Serialization/SerializedSILLoader.cpp#L36).  SerializedSILLoader::lookupSILFunction is how Differentiation and other passes (like inlining) find the implementation of SILFunctions.